commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareConfigServiceImpl.java

```diff
@@ -78,13 +78,42 @@ public CommcareConfigServiceImpl(@Qualifier("commcareAPISettings") SettingsFacad
     }
 
     @Override
-    public Config saveConfig(Config config) throws CommcareConnectionFailureException {
+    public Config saveUpdatedConfig(Config config, String oldName) throws CommcareConnectionFailureException {
```

Comment:
there is too much duplication between the methods - I might be wrong, but isn't saveNewConfig(config) the same as saveUpdatedConfig(config, config.getName()) ?